File: src/main/java/doom/despair/core/events/ValidEvents.java
package doom.despair.core.events;

import doom.despair.core.ClientGameEvent;
import doom.despair.core.ServerGameEvent;

import java.util.HashMap;
import java.util.Map;

public class ValidEvents {
    public static final Map<Integer, Class<? extends ClientGameEvent>> CLIENT_EVENTS = new HashMap<>();
    public static final Map<Integer, Class<? extends ServerGameEvent>> SERVER_EVENTS = new HashMap<>();
    public static void registerEvents() {
        CLIENT_EVENTS.put(0, ShipCreateEvent.ClientShipCreateEvent.class);



        SERVER_EVENTS.put(0, ShipCreateEvent.ServerShipCreateEvent.class);
    }
}

File: src/main/java/doom/despair/core/events/ValidEvents.kt
package doom.despair.core.events

import doom.despair.core.ClientGameEvent
import doom.despair.core.ServerGameEvent
import doom.despair.core.events.ShipCreateEvent.ClientShipCreateEvent
import doom.despair.core.events.ShipCreateEvent.ServerShipCreateEvent

object ValidEvents {
    val CLIENT_EVENTS: MutableMap<Int?, Class<out ClientGameEvent>> = HashMap()
    val SERVER_EVENTS: MutableMap<Int?, Class<out ServerGameEvent>> = HashMap()
    fun registerEvents() {
        CLIENT_EVENTS[0] = ClientShipCreateEvent::class.java
        SERVER_EVENTS[0] = ServerShipCreateEvent::class.java
    }
}
